.github/workflows/test.yml

```diff
@@ -0,0 +1,198 @@
+name: test
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '**'
+  pull_request:
+    branches:
+      - '**'
+  workflow_dispatch:
+
+
+concurrency:
+  # SHA is added to the end if on `main` to let all main workflows run
+  group: ${{ github.ref }}-${{ github.workflow }}-${{ github.event_name }}-${{ github.ref == 'refs/heads/main' && github.sha || '' }}
+  cancel-in-progress: true
+
+jobs:
+  configure:
+    name: Configure test matrix
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup Python environment
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+
+      - name: Find test paths
+        id: configure
+        run: |
+          python tests/build-job-matrix.py --per directory --verbose > matrix.json
+          cat matrix.json
+          echo ::set-output name=configuration::$(cat matrix.json)
+
+    outputs:
+      configuration: ${{ steps.configure.outputs.configuration }}
+
+  test:
+    name: ${{ matrix.os.emoji }} ${{ matrix.configuration.name }} - ${{ matrix.python.name }}
+    needs: configure
+    runs-on: ${{ matrix.os.runs-on }}
+    timeout-minutes: ${{ matrix.configuration.job_timeout }}
+    strategy:
+      fail-fast: false
+      matrix:
+        configuration: ${{ fromJson(needs.configure.outputs.configuration) }}
+        os:
+          - emoji: 🍎
+            matrix: macos
+            name: macOS
+            runs-on: macos-latest
+          - emoji: 🐧
+            matrix: ubuntu
+            name: Ubuntu
+            runs-on: ubuntu-latest
+#          - emoji: 🪟
+#            matrix: windows
+#            name: Windows
+#            runs-on: windows-latest
+        python:
```

Comment:
do python version of fun chibi emojis too? that would be great from a visual scanning perspective if they did.

Reply:
Could go with something below for the minor version if you think not-just-numerals looks better.  I think this might deal with the width, but I'm not sure it actually makes it better.

⑦ ⑧ ⑨ ⑩
❼ ❽ ❾ ❿

- 🍎❽
- 🍎❾
- 🍎❿

- 🍎❸❽
- 🍎❸❾
- 🍎❸❿

- 🍎3.8
- 🍎3.9
- 🍎3.10

I guess the full-width numerals are bigger and more visible?

- 🍎３.８
- 🍎３.９
- 🍎３.１０

Reply:
I'd prefer non-emoji numbers. Easier to read. Normal spacing seems fine to me

Reply:
normal numbers works for me as well. resolving.